server/src/test/java/org/elasticsearch/cluster/routing/GroupShardsIteratorTests.java

```diff
@@ -1,172 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the "Elastic License
- * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
- * Public License v 1"; you may not use this file except in compliance with, at
- * your election, the "Elastic License 2.0", the "GNU Affero General Public
- * License v3.0 only", or the "Server Side Public License, v 1".
- */
-
-package org.elasticsearch.cluster.routing;
-
-import org.apache.lucene.util.CollectionUtil;
-import org.elasticsearch.action.OriginalIndicesTests;
-import org.elasticsearch.action.search.SearchShardIterator;
-import org.elasticsearch.index.Index;
-import org.elasticsearch.index.shard.ShardId;
-import org.elasticsearch.test.ESTestCase;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
-
-import static org.elasticsearch.cluster.routing.ShardRoutingState.STARTED;
-
-public class GroupShardsIteratorTests extends ESTestCase {
```

Comment:
Outside of the two test utilities in this class, I don't believe there is any value in the tests for the methods we had. `totalSize` was test-only, the count 1 for empty method only had a single well-covered callsite in production and testing iteration and sorting is unnecessary for lists and covered by existing tests anyway.